pkg/services/login/authinfoservice/database/usagestats.go

```diff
@@ -0,0 +1,50 @@
+package database
+
+import (
+	"context"
+	"sync"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+var once sync.Once
+
+type loginStats struct {
+	DuplicateUserEntries int `xorm:"duplicate_user_entries"`
+}
+
+func (s *AuthInfoStore) GetLoginStats(ctx context.Context) (stats loginStats, err error) {
+	outerErr := s.sqlStore.WithDbSession(ctx, func(dbSession *sqlstore.DBSession) error {
+		rawSQL := `SELECT COUNT(*) FROM (` + s.duplicateUserEntriesSQL(ctx) + `) AS duplicate_user_entries`
+		_, err := dbSession.SQL(rawSQL).Get(&stats)
+		return err
+	})
+	if outerErr != nil {
+		return stats, outerErr
+	}
+	return stats, nil
+}
+
+func (s *AuthInfoStore) CollectLoginStats(ctx context.Context) (map[string]interface{}, error) {
```

Comment:
@sakjur refactored into a method on top of the AuthInfoStore

Reply:
👍